# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: 7e736b5732c1
# Full Hash: 7e736b5732c17f105f4692be0794568e23f88532
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-12 09:23:07
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 10 - CookieStore API - CHIP implementation shared between Document and CookieStore, r=cookie-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D220288
# ==============================================================================

diff -r 8620daea37ad -r 7e736b5732c1 dom/base/Document.cpp
--- a/dom/base/Document.cpp	Wed Sep 11 16:26:57 2024 +0000
+++ b/dom/base/Document.cpp	Wed Sep 11 16:26:58 2024 +0000
@@ -6653,8 +6653,9 @@
         continue;
       }
 
-      if (thirdParty && !CookieCommons::ShouldIncludeCrossSiteCookieForDocument(
-                            cookie, this)) {
+      if (thirdParty && !CookieCommons::ShouldIncludeCrossSiteCookie(
+                            cookie, CookieJarSettings()->GetPartitionForeign(),
+                            IsInPrivateBrowsing(), UsingStorageAccess())) {
         continue;
       }
 
@@ -6789,8 +6790,9 @@
                                                  nullptr, &thirdParty);
   }
 
-  if (thirdParty &&
-      !CookieCommons::ShouldIncludeCrossSiteCookieForDocument(cookie, this)) {
+  if (thirdParty && !CookieCommons::ShouldIncludeCrossSiteCookie(
+                        cookie, CookieJarSettings()->GetPartitionForeign(),
+                        IsInPrivateBrowsing(), UsingStorageAccess())) {
     return;
   }
 